tests/test_concurrency.py

```diff
@@ -25,11 +25,11 @@
 
     try:
         assert stream.is_connection_dropped() is False
-        assert stream.stream.get_extra_info("cipher", default=None) is None
+        assert stream.stream_writer.get_extra_info("cipher", default=None) is None
 
         stream = await backend.start_tls(stream, https_server.url.host, ctx, timeout)
         assert stream.is_connection_dropped() is False
-        assert stream.stream.get_extra_info("cipher", default=None) is not None
+        assert stream.stream_writer.get_extra_info("cipher", default=None) is not None
 
         await stream.write(b"GET / HTTP/1.1\r\n\r\n")
         assert (await stream.read(8192, timeout)).startswith(b"HTTP/1.1 200 OK\r\n")
```

Comment:
I debugged the failing test on 3.8, which was related to receiving empty bytes at this point. This seems to indicate the connection was dropped after the wirte, and by expanding assertions into…

```python
        await stream.write(b"GET / HTTP/1.1\r\n\r\n")
        assert stream.is_connection_dropped() is False
        data = await stream.read(8192, timeout)
        assert data
        assert data.startswith(b"HTTP/1.1 200 OK\r\n")
```

the first assertion is indeed failing: `stream.is_connection_dropped()` is actually `True`.

Reply:
Commenting all the other lines above in the `try:` block at least passes that assert, but then we still receive `b""` on the `.read()` call.

No idea what's happening. Might be a bug in 3.8rc1?